docs/design/cp-isolation-detection.md

```
@@ -0,0 +1,125 @@
+  `PeerAddressProvider` interface. `*peers.Peers` satisfies this interface
+  without modification — no changes needed in `cmd/main.go` or any consumer.
+- On upgrade, CP nodes gain the peer reachability check. A healthy,
+  well-connected CP node sees no behavioral difference because
+  `canReachAnyPeer()` succeeds immediately.
+
+## Time to Remediation
+
+With defaults (CheckInterval=15s, MaxErrorsThreshold=3, ApiServerTimeout=5s,
+peerDialTimeout=2s, MaxTimeForNoPeersResponse=30s):
+
+```
```

markdownlint-cli2 (0.23.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)